src/objects.rs

```diff
@@ -283,18 +279,20 @@ impl ObjectData {
             },
         });
 
-        let shape = shape.unwrap_or(ObjectShape::Rect { width, height });
-
         // Possibly copy properties from the template into the object
         // Any that already exist in the object's map don't get copied over
         if let Some(templ) = template {
+            shape.get_or_insert(templ.object.shape.clone());
```

Comment:
For performance reasons I guess it might make sense to use `get_or_insert_with` for all the cases where `clone()` needs to be called.